state/protocol/protocol_state/consumer.go

```diff
@@ -0,0 +1,23 @@
+package protocol_state
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+)
+
+// StateMachineEventsConsumer consumers notifications produced by OrthogonalStoreStateMachine instances. Any state machine
+// that performs processing of service events should notify the consumer about the events it received, successfully processed or
+// detected as invalid.
```

Comment:
```suggestion
// detected as invalid.
// Implementations must:
//   - be concurrency safe
//   - be non-blocking
//   - handle repetition of the same events (with some processing overhead).
```
Suggesting to copy API requirements from the HotStuff consumer